tlscanary/tools/one_crl_downloader.py

```diff
@@ -14,84 +14,61 @@
-            if subprocess.call(["git", "checkout", "-q", "master"], cwd=repo_dir) != 0:
-                logger.critical("Cannot checkout OneCRL-Tools git commit `master`")
-                sys.exit(5)
-        # `go get` internally uses `git pull`. `-d` prevents installation
-        logger.debug("Installing / updating Go package `%s`" % package)
-        if subprocess.call([go_bin, "get", "-u", "-d", package], env=go_env) != 0:
-            logger.critical("Cannot get Go package `%s`" % package)
-            sys.exit(5)
-        # Checkout a known-working commit before installation
-        logger.debug("Checking out commit `%s` in `%s`" % (commit, repo_dir))
-        if subprocess.call(["git", "checkout", "-q", commit], cwd=repo_dir) != 0:
-            logger.critical("Cannot checkout OneCRL-Tools git commit `%s`" % commit)
-            sys.exit(5)
-        # Install package
-        if subprocess.call([go_bin, "install", package], env=go_env) != 0:
-            logger.critical("Cannot install Go package `%s`" % package)
-            sys.exit(5)
-
-        # Run OneCRL Go binary to retrieve OnceCRL data
-        if sys.platform == "win32":
-            onecrl_bin = os.path.join(go_path, "bin", "oneCRL2RevocationsTxt.exe")
-        else:
-            onecrl_bin = os.path.join(go_path, "bin", "oneCRL2RevocationsTxt")
-        if not os.path.isfile(onecrl_bin):
-            logger.critical("Go package `oneCRL2RevocationsTxt` is missing executable")
-            sys.exit(5)
-        onecrl_cmd = [onecrl_bin, "--onecrlenv", onecrl_env]
-        logger.debug("Running shell command `%s`" % " ".join(onecrl_cmd))
-        try:
-            revocations_data = subprocess.check_output(onecrl_cmd, env=go_env)
-        except subprocess.CalledProcessError as error:
-            logger.critical("Could not fetch revocations data: %s" % error)
-            sys.exit(5)
-
-        # oneCRL2RevocationsTxt does not indicate failure, but the result is empty.
-        # Can we be sure this can never happen during regular operation?
-        # See https://github.com/mozmark/OneCRL-Tools/issues/3
-        if len(revocations_data) == 0:
-            logger.critical("Revocations data was empty. Likely network failure.")
-            sys.exit(5)
-
-        # Write OneCRL data to cache
-        logger.debug("Caching OneCRL revocations data in `%s`" % cache_file)
-        with open(cache_file, "wb") as f:
-            f.write(revocations_data)
-
-    else:
-        logger.warning("Using cached OneCRL revocations data from `%s`" % cache_file)
-
-    return cache_file
+    if cache_id in dc:
+        logger.warning("Using cached OneCRL cert_storage data from `%s`" % dc[cache_id])
+        return
+
+    if _ONE_CRL_TOOLS_REPO not in dc:
+        subprocess.call(["git", "clone", _ONE_CRL_TOOLS_GIT_URL, dc[_ONE_CRL_TOOLS_REPO]])
+
+    cached_security_state = dc[cache_id]
+    os.makedirs(cached_security_state)
+
+    repo_dir = dc[_ONE_CRL_TOOLS_REPO]
```

Comment:
I though that it may be a better idea to place our copy of the OneCRL repo in the cache rather than bank on the repo that happens to exist within $GOPATH, especially since this particular procedure is not dependent upon Go at all and as such shouldn't hope that it's there.